main.cpp

```diff
@@ -1 +1,310 @@
+// int StringListIndexOf(char** list, char* str);        
+int StringListIndexOf(char** list, const char* str)
+{
+    if(!list || !str[0]) return -2;
+    
+    int counter(0);
+    
+    while(list!=NULL)
+    {
+        if(!strcmp(list[0], str)) return counter;
+        counter++;
+        list = (char**)list[1];
+    }
+
+    return -1;
+}
+
+// Removes all occurrences of str in the list.
+// void StringListRemove(char*** list, String str);
+void StringListRemove(char*** list, const char* str)    
+{
+    if(!*list || !str[0]) return;
+
+    char** current(*list), **previous(NULL);
+  
+    while (current!=NULL)
+    {
+        if(current == *list && !strcmp(current[0], str))
+        {
+            *list = (char**)current[1];
+            free(current[0]);
+            free(current);
+            if(!*list) return;
+            current = *list;
+            
+        }
+
+        else if(!strcmp(current[0], str))
+        {
+            previous[1] = current[1];
+            free(current[0]);
+            free(current);
+            current = (char**)previous[1];
+        }
+
+        else 
+        {  
+            previous = current;
+            current = (char **)current[1];
+        }
+        
+    }
+
+}
+
+// Removes duplicate entries from the list.
+// void StringListRemoveDuplicates(char** list);
+void StringListRemoveDuplicates(char** list)
+{
+    if(!list) return;
+    char** current(list), **previous(NULL);
+
+    while(current!=NULL)
+    {
+        char** next((char**)current[1]);
+
+        StringListRemove(&next, current[0]);
+        previous = current;
+        previous[1] = (char*) next;
+        current = next;
+    }
+}
+
+// Replaces every occurrence of the before, in each of the string lists's strings, with after.
+// void StringListReplaceInStrings(char** list, char* before, char* after);
+void StringListReplaceInStrings(char** list, const char* before, const char* after)
+{
+    if(!list || !before[0] || !after[0]) return;
+
+    char** current(list);
+
+    while (current!=NULL)
+    {
+        if (!strcmp(current[0], before))
+        {
+            char* tmp = (char*)realloc(current[0], strlen(after)+1);
+
+            if (tmp)
+            {
+                current[0] = tmp;
+                strcpy(current[0], after);
+            }
+            
+            else return;
+        }
+        
+        current = (char **)current[1];
+    }
+
+}
+
+// Sorts the list of strings in ascending order
+// void StringListSort(char** list);
+void StringListSort(char** list)
+{
+    if(!list) return;
+
+    char** left(list), **right((char**)list[1]);
+
+    while(left[1])
+    {   
+        while(right)
+        {
+            char rightLower [strlen(right[0]) + 1];
+            char leftLower [strlen(left[0]) + 1];
+
+            for (int i = 0; i <= strlen(left[0]); i++)
+            {
+                leftLower[i] = tolower(left[0][i]);
+            }
+
+            for (int i = 0; i <= strlen(right[0]); i++)
```

Comment:
The same is here. Decreased performance. Calls to the 'strlen' function have being made multiple times when a condition for the loop's continuation was calculated.